File: proj 1/Packet.py
class Packet:
	def __init__(self, src_port, dest_port, seq_num, ack_num, flags, data, checksum = None):
		self.src_port = src_port & 0xffff
		self.dest_port = dest_port & 0xffff
		self.seq_num = seq_num
		self.ack_num = ack_num
		self.data = data
		self.ACK = flags[0]
		self.PSH = flags[1]
		self.RST = flags[2]
		self.SYN = flags[3]
		self.FIN = flags[4]

File: Packet.java
public class Packet {

    int seqNum;
    boolean isAcked;
    byte[] data;
    int resentCounter;


    public Packet(seqNum, isAcked, data, resentCounter) {
        this.seqNum = seqNum;
        this.isAcked = isAcked;
        this.data = data;
        this.resentCounter = resentcounter;
    }

    public void setSeqNum(int seqNum) {
        this.seqNum = seqNum;
    }

    public void setAck(boolean isAcked) {
        this.isAcked = isAcked;
    }
}
